### PR TITLE
Add num_cpus example

### PR DESCRIPTION
**Pull request by V1shvesh**
@budziq have a look
**Edit:** fixes #307

### CODE REVIEW

File: src/basics.md [hunk 1]
```diff
@@ -847,6 +848,31 @@ fn run() -> Result<()> {
...
+[![num_cpus-badge]][num_cpus] [![cat-hardware-support-badge]][cat-hardware-support]
+
+Counts the number of logical cpu cores in current machine.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate num_cpus;
+#
+# error_chain! {
```
Comment (budziq) [inline src/basics.md]: No need for `error_chain` as there is no possibility for error handling in this example. Lets just remove it.

File: src/basics.md [hunk 2]
```diff
@@ -847,6 +848,31 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-check-cpu-cores]: #ex-check-cpu-cores
+<a name="ex-check-cpu-cores"></a>
+## Check number of logical cpu cores
+
+[![num_cpus-badge]][num_cpus] [![cat-hardware-support-badge]][cat-hardware-support]
+
+Counts the number of logical cpu cores in current machine.
```
Comment (budziq) [inline src/basics.md]: I would rephrase that with something along the lines of:
"Shows the number of logical cpu cores with help of `num_cpus::get`" and I would hyperlink the symbol reference to its respective docs.rs

**Comment by budziq** [comment]
also please refer the original issue in your PR text like "fixes #307" for the issue to autoclose on this PR being merged

File: src/basics.md (hunk 2 above)
Comment (V1shvesh) [reply src/basics.md]: I hope the heading is fine as it is.

**Comment by budziq** [comment]
merged with some minor cleanup. Thanks @V1shvesh

**Comment by V1shvesh** [comment]
@budziq I'm sorry for asking, but I'm seeing "closed pull request" instead of "merged pull request". This is fine, or not?

**Comment by budziq** [comment]
Yeah, I've pulled your changes and done some minor finish line cleaning myself and pushed them directly. Github is not able to handle this correctly so one has to close the PR by hand.

You can see your example in the gitlog (your authorship preserved) and published in rendered cookbook now :+1:

**Comment by V1shvesh** [comment]
Oh, that's not a problem. My main concern is would this count as a PR for hacktoberfest?

**Comment by budziq** [comment]
No idea, sorry! But you can ask digital ocean directly. They are usually quite responsive.
